lib/yang_wrappers.c

```diff
@@ -1127,3 +1133,31 @@ void yang_str2mac(const char *value, struct ethaddr *mac)
 {
     (void)prefix_str2mac(value, mac);
 }
+
+char *yang_nexthop_type2str(char *type_str, int len, uint32_t ntype)
+{
+	switch (ntype) {
+	case NEXTHOP_TYPE_IFINDEX:
+		strlcpy(type_str, "ifindex", len);
+		break;
+	case NEXTHOP_TYPE_IPV4:
+		strlcpy(type_str, "ip4", len);
+		break;
+	case NEXTHOP_TYPE_IPV4_IFINDEX:
+		strlcpy(type_str, "ip4-ifindex", len);
+		break;
+	case NEXTHOP_TYPE_IPV6:
+		strlcpy(type_str, "ip6", len);
+		break;
+	case NEXTHOP_TYPE_IPV6_IFINDEX:
+		strlcpy(type_str, "ip6-ifindex", len);
+		break;
+	case NEXTHOP_TYPE_BLACKHOLE:
+		strlcpy(type_str, "blackhole", len);
```

Comment:
Do we need a copy here or can we return the value as is as const char *.

Reply:
Agreed, returning a `const char *` would be better here.

I only think this function doesn't belong in this file. If we add a function like this for every YANG enumeration, this isn't going to scale (also, we don't have access to daemons-specific enums in libfrr).

In general we can use libyang to convert enum values to their equivalent YANG text representation. We do that in the `yang_data_new_enum()` function, but it does that using a `while` loop so it's not super performant. Depending on what you're trying to do it might be sufficient though.

One thing we could do is to ask the libyang maintainers if they can provide a new API to provide faster enum value to text mappings. If we have that we won't need to write and maintain several mapping functions like this ourselves.

Reply:
@rwestphal  Agreed if libayng can provide enum2str it would be ideal.
Until then we would need to use custom wrapper. 
I agree the API should not be part of yang_wrapper.c file. But nexthop enum can be used my multiple modules so resided in the common yang file.
But once libyang provides a wrapper we can remove this api.

Reply:
Sounds fine to me.

Reply:
In that case, please make this function return `const char *` as Santosh suggested.

Reply:
Ack. I have incorporated the comment and added `const char *`